pkg/api/types.go

```diff
@@ -1,38 +1,65 @@
+	ServiceCIDR                         string             `json:"serviceCIDR"`
+	NamedCerts                          []NamedCert        `json:"namedCerts,omitempty"`
+	PodCIDR                             string             `json:"podCIDR"`
+	ReleaseImage                        string             `json:"releaseImage"`
+	APINodePort                         uint               `json:"apiNodePort"`
+	IngressSubdomain                    string             `json:"ingressSubdomain"`
+	OpenShiftAPIClusterIP               string             `json:"openshiftAPIClusterIP"`
+	ImageRegistryHTTPSecret             string             `json:"imageRegistryHTTPSecret"`
+	RouterNodePortHTTP                  string             `json:"routerNodePortHTTP"`
+	RouterNodePortHTTPS                 string             `json:"routerNodePortHTTPS"`
+	OpenVPNNodePort                     string             `json:"openVPNNodePort"`
+	BaseDomain                          string             `json:"baseDomain"`
+	NetworkType                         string             `json:"networkType"`
+	Replicas                            string             `json:"replicas"`
+	EtcdClientName                      string             `json:"etcdClientName"`
+	OriginReleasePrefix                 string             `json:"originReleasePrefix"`
+	OpenshiftAPIServerCABundle          string             `json:"openshiftAPIServerCABundle"`
+	CloudProvider                       string             `json:"cloudProvider"`
+	CVOSetupImage                       string             `json:"cvoSetupImage"`
+	InternalAPIPort                     uint               `json:"internalAPIPort"`
+	RouterServiceType                   string             `json:"routerServiceType"`
+	KubeAPIServerResources              []ResourceRequests `json:"kubeAPIServerResources"`
+	OpenshiftControllerManagerResources []ResourceRequests `json:"openshiftControllerManagerResources"`
+	ClusterVersionOperatorResources     []ResourceRequests `json:"clusterVersionOperatorResources"`
+	KubeControllerManagerResources      []ResourceRequests `json:"kubeControllerManagerResources"`
+	OpenshiftAPIServerResources         []ResourceRequests `json:"openshiftAPIServerResources"`
+	KubeSchedulerResources              []ResourceRequests `json:"kubeSchedulerResources"`
+	CAOperatorResources                 []ResourceRequests `json:"cAOperatorResources"`
+	OAuthServerResources                []ResourceRequests `json:"oAuthServerResources"`
+	ClusterPolicyControllerResources    []ResourceRequests `json:"clusterPolicyControllerResources"`
+	AutoApproverResources               []ResourceRequests `json:"autoApproverResources"`
+	OpenVPNClientResources              []ResourceRequests `json:"openVPNClientResources"`
+	OpenVPNServerResources              []ResourceRequests `json:"openVPNServerResources"`
 }
 
 type NamedCert struct {
 	NamedCertPrefix string `json:"namedCertPrefix"`
 	NamedCertDomain string `json:"namedCertDomain"`
 }
+
+type ResourceRequests struct {
```

Comment:
nit: s/ResourceRequests/ResourceRequirements/ as in API (https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2087)